akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCart.scala

```diff
@@ -0,0 +1,191 @@
+package sample.cqrs
+
+import java.time.Instant
+
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.cluster.sharding.typed.scaladsl.ClusterSharding
+import akka.cluster.sharding.typed.scaladsl.Entity
+import akka.cluster.sharding.typed.scaladsl.EntityTypeKey
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.scaladsl.Effect
+import akka.persistence.typed.scaladsl.EventSourcedBehavior
+import akka.persistence.typed.scaladsl.ReplyEffect
+
+object ShoppingCart {
+
+  /**
+   * The current state held by the persistent entity.
+   */
+  final case class State(items: Map[String, Int], checkedOut: Boolean) {
```

Comment:
```suggestion
  final case class State(items: Map[String, Int], checkedOut: Boolean) extends CborSerializable {
```

Should this sample also demo snapshotting or is it too out of scope?